candle-nn/src/batch_norm.rs

```diff
@@ -34,79 +42,108 @@ impl From<f64> for BatchNormConfig {
             eps,
             remove_mean: true,
             affine: true,
+            momentum: 0.1,
         }
     }
 }
 
-#[derive(Clone, Debug)]
+#[derive(Debug)]
 pub struct BatchNorm {
-    running_mean: Tensor,
-    running_var: Tensor,
+    running_mean: Var,
+    running_var: Var,
     weight_and_bias: Option<(Tensor, Tensor)>,
     remove_mean: bool,
     eps: f64,
-    num_features: usize,
+    momentum: f64,
 }
 
 impl BatchNorm {
     pub fn new(
-        num_features: usize,
         running_mean: Tensor,
         running_var: Tensor,
         weight: Tensor,
         bias: Tensor,
         eps: f64,
+        momentum: f64,
```

Comment:
Let's have a separate function to create a batch-norm with momentum and use the default value here so as not to break the current api (also most use cases don't care much about the momentum value anyway).

Reply:
Okay, I've added `with_momentum` variants of the `new` methods. What are your thoughts on my removing of the `num_features` parameter from the `new` methods? It does simplify things, but is still breaking. These changes will break some code due to the changes to `BatchNormConfig`.

Reply:
Would prefer not breaking the api and so adding `num_features` back.

Reply:
I've added it back, and modified `check_validity` to use it.